internal/data_source_dnssec.go

```diff
@@ -152,9 +154,18 @@ func (d *dnsSecDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	zoneId := int32(getZoneId.ValueInt64())
 
-	getDnsSec, response, err := d.client.idnsApi.DNSSECAPI.GetZoneDnsSec(ctx, zoneId).Execute() //nolint
+	zoneID32, err := utils.CheckInt64toInt32Security(getZoneId.ValueInt64())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error before Overflow",
```

Comment:
`Error: value exceeds the valid range`

I think the message above represent the problem better